cmd/helm/repo_add.go

```diff
@@ -131,6 +135,23 @@ func addRepository(name, url, username, password string, home helmpath.Home, cer
 		return fmt.Errorf("Looks like %q is not a valid chart repository or cannot be reached: %s", url, err.Error())
 	}
 
+	// Lock the repository file for concurrent processes synchronization and re-read its content before updating it
+	fileLock := flock.New(home.RepositoryFile())
+	lockCtx, cancel := context.WithTimeout(context.Background(), 5*time.Minute)
+	defer cancel()
+	locked, err := fileLock.TryLockContext(lockCtx, time.Second)
+	if err == nil && locked {
+		defer fileLock.Unlock()
+	}
+	if err != nil {
+		return err
+	}
+
+	f, err = repo.LoadRepositoriesFile(home.RepositoryFile())
```

Comment:
It might be worth putting a comment on why `LoadRepositoriesFile` is being run a second time. I see it because I read another conversation on it recently. May not be obvious to everyone or even me in the future.